radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/forks/ForkOverwritesFromPropertiesModule.java

```diff
@@ -69,52 +69,67 @@
 import com.google.inject.TypeLiteral;
 import com.google.inject.multibindings.OptionalBinder;
 import com.radixdlt.properties.RuntimeProperties;
+import java.util.Optional;
 import java.util.Set;
 import java.util.function.UnaryOperator;
 import java.util.stream.Collectors;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 public class ForkOverwritesFromPropertiesModule extends AbstractModule {
+  private static final String PROPERTIES_PREFIX = "overwrite_forks.";
+
   private static final Logger logger = LogManager.getLogger();
 
-  private static class ForkOverwrite implements UnaryOperator<Set<ForkConfig>> {
+  private static class ForkOverwrite implements UnaryOperator<Set<ForkBuilder>> {
     @Inject private RuntimeProperties properties;
 
     @Override
-    public Set<ForkConfig> apply(Set<ForkConfig> forkConfigs) {
-      return forkConfigs.stream()
+    public Set<ForkBuilder> apply(Set<ForkBuilder> forkBuilders) {
+      return forkBuilders.stream()
           .map(
               c -> {
-                var epochOverwrite = properties.get("overwrite_forks." + c.name() + ".epoch", "");
+                final var forkDisabledOverwrite =
```

Comment:
I think this lambda worth moving into dedicated method and refactoring. Too much low-level logic in one method, IMO.